javascript/javascript3/week3/shopping/shopping.js

```diff
@@ -0,0 +1,280 @@
+  }
+
+  searchProduct(productName) {
+    //should return an array of product that match the productName parameter
+    const matchedProducts = this.products.filter(
+      (cProduct) => cProduct.name === productName
+    );
+    return matchedProducts;
+  }
+
+  getTotal() {
+    //should get the total price of the products in the shoppingcart.
+    const reducer = (accmulator, cProduct) => accmulator + cProduct.price;
+    const total = this.products.reduce(reducer, 0);
+    return total;
+  }
+
+  renderProducts(userName) {
+    const user = document.querySelector("#user_name");
+    user.textContent = userName;
+    this.products.forEach((product) => {
+      const ul = document.createElement("ul");
+      ul.className = "detail_list";
+      const productLi = document.createElement("li");
+      const priceLi = document.createElement("li");
+      const name = product.name;
+      const price = product.price;
+      productLi.textContent = name;
+      priceLi.textContent = price;
+      ul.appendChild(productLi);
+      ul.appendChild(priceLi);
+      const li = document.createElement("li");
+      li.appendChild(ul);
+      mainUL.appendChild(li);
+    });
+    const ul = document.createElement("ul");
+    ul.className = "detail_list";
+    const totalpriceLabelLi = document.createElement("li");
+    const totalPriceLi = document.createElement("li");
+    totalpriceLabelLi.textContent = "Total Price: ";
+    totalPriceLi.textContent = this.getTotal();
+    ul.appendChild(totalpriceLabelLi);
+    ul.appendChild(totalPriceLi);
+    const li = document.createElement("li");
+    li.appendChild(ul);
+    mainUL.appendChild(li);
+  }
+
+  getUser() {
+    return fetch(" https://jsonplaceholder.typicode.com/users/1")
+      .then((response) => response)
+      .catch((e) => e);
+  }
+}
+//Part 2 create instance of ShoppingCart
+const shoppingCart = new ShoppingCart();
+//create instances of Product
+const flatscreen = new Product("Flat-screen", 5000);
+const radio = new Product("Radio", 2500);
+const laptop = new Product("Laptop", 6000);
+const iphone = new Product("Apple iphone", 6030);
+const glaxy = new Product("Samsung Glaxy", 60400);
+const camra = new Product("Sony Camra", 63000);
+//add products in shopingcart
+shoppingCart.addProduct(flatscreen);
+shoppingCart.addProduct(radio);
+shoppingCart.addProduct(laptop);
+shoppingCart.addProduct(iphone);
+shoppingCart.addProduct(glaxy);
+shoppingCart.addProduct(camra);
+//get user name then render products from shopping cart
+shoppingCart
+  .getUser()
+  .then((response) => response.json())
+  .then((user) => {
+    shoppingCart.renderProducts(user.name);
+  })
+  .catch((e) => console.log(e));
+
+//Part 3 Assuming dkk as default currency
+const plant = new Product("plant", 50);
+const currencyName = "GBP";
+plant
+  .convertToCurrency(currencyName)
+  .then((price) =>
+    console.log(`${plant.name}'s price in ${currencyName} is : ${price}`)
+  );
+const wrongCurrencyName = "Wrong Currency";
```

Comment:
this code `const wrontCurrencyName ...` will execute _before_ .then() above,
convert to async / await.

Reply:
Thank you for your precious time for the review and valuable comments. I have implemented all the suggested changes.